### DISCUSSION TITLE
RY coding

### DISCUSSION THREAD

**Discussion by strewick** [iqtree/iqtree2#210] [Q&A] [answered]
Third codon RY recoding has been discussed previously. A suggestion was to use binary 01 coding instead of RY for purine/pyrimadine nucleotides. iqtree recognises binary states and invokes suitable models.
HOWEVER, if the input data are sequences where part of the data remain  four state and part two state (=RY coding as 01), does this require a partition file to invoke DNA model for part of the data. Without partition file I find that 'morphological' data is identified by iqtree, which appears to exclude DNA models.

What is the simplest way to apply DNA and BIN models to partitioned data of this sort?

**Comment by bqminh** [comment]
It's answered here: http://www.iqtree.org/doc/Frequently-Asked-Questions#can-i-mix-dna-and-protein-data-in-a-partitioned-analysis

**Reply by strewick** [reply]
Yes I have seen that information, however I am trying to splice together that information with your previous response to a question about RY coding. https://groups.google.com/g/iqtree/c/Yoh47UYBhiw

Am I to take it that in order to run analysis with a) DNA sequences (codons 1 and 2), and b) 0,1 coded 3rd codon positions they have to be in separate data files?

**Reply by bqminh** [reply, marked as answer]
You can keep the DNA alignment untouched, and pull all 3rd codon positions to another file with 0/1 recoding, say `bin.phy`. Then prepare a partition file (e.g. partition.nex):
```
#nexus
begin sets;
  charset part1 = dna.phy: 1-300\3; [ replace 300 with the actual alignment length ]
  charset part2 = dna.phy: 2-300\3;
  charset part3 = bin.phy;
end;
```

Then run something like:
```
iqtree2 -p partition.nex
```
